### ISSUE TITLE
[WinUI] [UIKit] CollectionView Items do not resize if the inner content changes.

### ISSUE THREAD

**Issue by drasticactions** [dotnet/maui#13451]
### Description

If you modify an existing CollectionView item on iOS, Catalyst, or WinUI the outer cell does not resize to reflect the new content. This should similar, if not the same, as https://github.com/dotnet/maui/issues/12118, but with a simple reproduction step (If that's the case, IMO, I would close that issue and redirect it to this one, since it's broader in scope).

For WinUI in my sample, the content doesn't directly overlap, rather there are gaps present between the cells as you add or remove them. With enough changes of the inner cells and their orders, you can cause the contents to overlap.

https://user-images.githubusercontent.com/898335/220037568-884189f6-ff99-4169-a663-a4e7712871d4.mp4

### Steps to Reproduce

1. Clone https://github.com/drasticactions/MauiRepros/tree/main/MauiCollectionViewResize
2. Run the sample

### Link to public reproduction project repository

https://github.com/drasticactions/MauiRepros/tree/main/MauiCollectionViewResize

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Windows, macOS

### Affected platform versions

All affected platform versions.

### Did you find any workaround?

Somehow force the window or control to resize. On Windows/Mac you can resize the window, but that's not possible on iPhone.

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with 17.6 Preview (17.6 build 1539). Can repro on macOS with sample project.
[MauiRepros-main.zip](https://github.com/dotnet/maui/files/11699555/MauiRepros-main.zip)
![13451 macOS](https://github.com/dotnet/maui/assets/77434865/3a76f4d1-c021-447c-acdd-a3d0e8a4cbf5)

**Comment by eengstroem** [comment]
I believe I'm experiencing the same issue [here](https://github.com/dotnet/maui/issues/16210). It seems to work with android. The link has a link to a reproduction repo.

**Comment by Baraiboapex** [comment]
Is there a way to force the whole outer cell of the collection view to redraw via triggering invalidation as shown in the workaround that @Pughgramming pointed out?

**Comment by konradzuse** [comment]
In our case we're loading images asynchronously, so the cell resizes only once the images have been populated. The only workaround is to rotate the device so that the cell is redrawn.

I've just updated to 8.0.60 but the issue remains.

**Comment by Baraiboapex** [comment]
@konradzuse 

Yeah, I did the same thing for our application. and its still not working either. Here is a link to a repo that I am using to reproduce this problem on iOS, btw: https://github.com/Baraiboapex/Armory2/tree/master/Armory2

**Comment by Baraiboapex** [comment]
Does anybody know of any libraries that can be used in place of this?

**Comment by HarishwaranVijayakumar** [comment]
This issue is not reproducible in the latest .NET MAUI version 9.0.90 on windows and mac in both CV1 and CV2. I have attached video for your reference.
| Mac| Windows |
|----------|----------|
| <video src="https://github.com/user-attachments/assets/cf6ab4d9-8f48-4cee-95dd-7e0599b16005"> | <video src="https://github.com/user-attachments/assets/966eaa52-96ad-466d-a0a9-3365952899df"> |
